src/Kucipong/Spock/ReqParam.hs

```diff
@@ -0,0 +1,42 @@
+{-|
+Module      : Kucipong.Spock.ReqParam
+Description : Helper functions for request parameters.
+Stability   : experimental
+Portability : POSIX
+-}
+
+module Kucipong.Spock.ReqParam where
+
+import Kucipong.Prelude
+
+import Control.FromSum ( fromEitherM )
+import Data.HVect ( HasRep, HVectElim )
+import Network.Wai ( requestBody )
+import Web.FormUrlEncoded ( FromForm(fromForm), toForm, urlDecodeAsForm )
+import Web.Spock ( ActionCtxT, Path, body, post, redirect, renderRoute, root )
+import Web.Spock.Core ( SpockCtxT, request, params )
+
+getReqParam
+    :: forall ctx m a
+     . (FromForm a, MonadIO m)
+    => ActionCtxT ctx m a
+getReqParam = getReqParamErr handleFormDecodeError
+  where
+    handleFormDecodeError :: Text -> ActionCtxT ctx m a
+    handleFormDecodeError err = do
+        -- TODO: How should we handle this error?
```

Comment:
Please make it possible to render any page template we want with errors as template parameters.
I'll implement later to show the errors as pop up message on the page.